crates/lib3h/src/engine/space_layer.rs

```diff
@@ -325,7 +325,21 @@ impl<'engine> GhostEngine<'engine> {
                     );
                 }
             }
-            _ => panic!("can't handle space layer receive of {:?}", p2p_msg),
+            P2pProtocol::Gossip(gossip_data) => {
+                let remote_gossip = RemoteGossipBundleData {
+                    from_peer_name: gossip_data.from_peer_name.clone(),
+                    bundle: gossip_data.bundle.clone(),
+                };
+                let space_gateway =
+                    self.get_space(&gossip_data.space_address, &gossip_data.to_peer_name.into())?;
+                space_gateway.publish(
+                    span.follower("TODO"),
+                    GatewayRequestToChild::Dht(DhtRequestToChild::HandleGossip(remote_gossip)),
+                )?;
+            }
+            _ => {
+                panic!("can't handle space layer receive of {:?}", p2p_msg);
+            }
```

Comment:
Was this the critical missing piece for the test to pass?

Reply:
A half of it yes. The other half is having gateway bubble up the dht request to engine.